include/sparrow/dynamic_bitset.hpp

```diff
@@ -0,0 +1,338 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cassert>
+#include <climits>
+
+#include "sparrow/buffer.hpp"
+
+namespace sparrow
+{
+    template <class B>
+    class dynamic_bitset_base
+    {
+    public:
+
+        using self_type = dynamic_bitset_base<B>;
+        using storage_type = B;
+        using block_type = typename storage_type::value_type;
+        using value_type = bool;
+        using size_type = typename storage_type::size_type;
+
+        size_type size() const noexcept;
+        size_type null_count() const noexcept;
+
+        bool test(size_type pos) const;
```

Comment:
`test` and `set` are not `noexcept`. Is that on purpose?

Reply:
Yes, these are the equivalent of `operator[]` of a container like `std::vector`. I did not implement such operators because these would require other reference proxies that we don't really need for now.